code/modules/mob/living/brain/living_brain.dm

```diff
@@ -18,31 +18,14 @@
 	return ..()
 
 /mob/living/brain/say_understands(other)//Goddamn is this hackish, but this say code is so odd
-	if(isAI(other))
-		if(!(container && istype(container, /obj/item/mmi)))
-			return 0
-		else
-			return 1
-	if(istype(other, /mob/living/silicon/decoy))
-		if(!(container && istype(container, /obj/item/mmi)))
-			return 0
-		else
-			return 1
-	if(ispAI(other))
-		if(!(container && istype(container, /obj/item/mmi)))
-			return 0
-		else
-			return 1
-	if(isrobot(other))
-		if(!(container && istype(container, /obj/item/mmi)))
-			return 0
-		else
-			return 1
+	if(!ismob(other))
+		return ..()
+	if(issilicon(other))
+		return container && istype(container, /obj/item/mmi)
```

Comment:
Looking at this again, we can simplify a bit further, since istype already does a null check:
```suggestion
		return istype(container, /obj/item/mmi)
```